src/events/eventActions.js

```diff
@@ -73,7 +72,6 @@ export const startEventPolling = (queueId: number, eventId: number) => async (
 
       const actions = eventsToActions(getState(), events);
 
-      actionCreator(dispatch, actions, getState());
       dispatchOrBatch(dispatch, actions);
 
       actions.forEach(action => {
```

Comment:
> If later we have some that need to come before and others after for some reason, we can have two different functions handling the different side effects

Or, I guess, move these side effects into a new custom Redux Middleware; in there, we can choose when to call `next`.

Reply:
> This has a subtle effect on behavior which it's not obvious to me whether it matters.

Hmmm, right. I assumed it wouldn't matter, because `clearTypingNotification` starts off by doing nothing for 15 seconds. But there's another thing to think about: `clearTypingNotification` isn't dispatched unconditionally:

```
    case EVENT_TYPING_START:
      if (Object.keys(state.typing).length === 0) {
        dispatch(clearTypingNotification());
      }
      break;
```

I'll try and see if I can understand what's going on with that condition; it seems kind of wrong, like "If there are no typing notifications, clear the typing notifications".

Reply:
OK, I resolved this confusion by reimplementing the client-side timer for the typing state; take a look, and see what you think.